[Failing test] SystemLogMonitor should generate node condition and events for corresponding errors

**Which jobs are failing**:
`ci-npd-e2e-node`

**Which test(s) are failing**:
`SystemLogMonitor should generate node condition and events for corresponding errors`

**Since when has it been failing**:
Last successful run (23 Oct 2020, 10:07 PDT): https://prow.k8s.io/view/gcs/kubernetes-jenkins/logs/ci-npd-e2e-node/1319686617375444992
First unsuccessful run (23 Oct 2020, 12:08 PDT): https://prow.k8s.io/view/gcs/kubernetes-jenkins/logs/ci-npd-e2e-node/1319717073798041600

**Testgrid link**:
https://testgrid.k8s.io/sig-node-node-problem-detector#ci-npd-e2e-node

**Reason for failure**:
```
_output/local/go/src/k8s.io/kubernetes/test/e2e_node/node_problem_detector_linux.go:245
Timed out after 60.000s.
Expected success, but got an error:
    <*errors.errorString | 0xc001212cd0>: {
        s: "node condition \"TestCondition\" not found",
    }
    node condition "TestCondition" not found
_output/local/go/src/k8s.io/kubernetes/test/e2e_node/node_problem_detector_linux.go:360
```

**Anything else we need to know**:
[pull-npd-e2e-node](https://prow.k8s.io/job-history/gs/kubernetes-jenkins/pr-logs/directory/pull-npd-e2e-node) presubmit job in k8s.io/node-problem-detector repository also started to fail in the same fashion.

Attempted to debug it myself, found out the following stuff:
* last successful run was accessing the apiserver via HTTP:
```
W1023 17:12:40.514888    2808 test_context.go:420] Unable to find in-cluster config, using default host : http://127.0.0.1:8080
```
* all unsuccessful runs attempted to access the apiserver via HTTPS but to no avail:
```
W1023 19:14:09.903857     995 test_context.go:437] Unable to find in-cluster config, using default host : https://127.0.0.1:6443
(...)
W1023 19:15:40.789213     925 util.go:106] Health check on "https://127.0.0.1:6443/healthz" failed, status=500
```
* perused the history of `k/k/test` to find out that the PR https://github.com/kubernetes/kubernetes/pull/94723 is almost certainly the reason of the failures (or rather uncovered some problem with the NPD or its tests):
  * first, it coincides with the start of the failures of `ci-npd-e2e-node`,
  * second, we changed the default hostname of the apiserver in node tests in that PR.

Judging by the code and the changes in the aforementioned PR I think that a quick mitigation would be to add a `--host=http://127.0.0.1:8080` flag to the `--node-test-args` of these jobs.

/assign
/sig node
/cc @knight42 @Random-Liu

@knight42 @Random-Liu the submission of #96050 hasn't fixed the problems, unfortunately :/.

Sample failing run: https://prow.k8s.io/view/gcs/kubernetes-jenkins/logs/ci-npd-e2e-node/1324222964127764480

We still have some peculiar problem related to health checking:
```
W1105 05:45:46.863813     983 util.go:106] Health check on "https://127.0.0.1:6443/healthz" failed, status=500
```

and now NPD panics, too:
```
panic: open /var/run/secrets/kubernetes.io/serviceaccount/token: no such file or directory
goroutine 1 [running]:
k8s.io/node-problem-detector/pkg/exporters/k8sexporter/problemclient.NewClientOrDie(0xc0003eb790, 0xc0000b6d00, 0x0)
	/home/prow/go/src/k8s.io/node-problem-detector/pkg/exporters/k8sexporter/problemclient/problem_client.go:73 +0x3f7
k8s.io/node-problem-detector/pkg/exporters/k8sexporter.NewExporterOrDie(0xc0003eb790, 0xc000492370, 0x1)
	/home/prow/go/src/k8s.io/node-problem-detector/pkg/exporters/k8sexporter/k8s_exporter.go:52 +0x4f
main.main()
	/home/prow/go/src/k8s.io/node-problem-detector/cmd/nodeproblemdetector/node_problem_detector.go:60 +0x13a
```

--- a/test/e2e_node/BUILD
+++ b/test/e2e_node/BUILD
@@ -67,7 +67,6 @@ go_library(
     ] + select({
         "@io_bazel_rules_go//go/platform:android": [
             "//pkg/util/procfs:go_default_library",
-            "//staging/src/k8s.io/api/rbac/v1:go_default_library",
             "//staging/src/k8s.io/apimachinery/pkg/fields:go_default_library",
             "//staging/src/k8s.io/apimachinery/pkg/labels:go_default_library",
             "//staging/src/k8s.io/apimachinery/pkg/types:go_default_library",
@@ -87,7 +86,6 @@ go_library(
         ],
         "@io_bazel_rules_go//go/platform:linux": [
             "//pkg/util/procfs:go_default_library",
-            "//staging/src/k8s.io/api/rbac/v1:go_default_library",
             "//staging/src/k8s.io/apimachinery/pkg/fields:go_default_library",
             "//staging/src/k8s.io/apimachinery/pkg/labels:go_default_library",
             "//staging/src/k8s.io/apimachinery/pkg/types:go_default_library",

--- a/test/e2e_node/node_problem_detector_linux.go
+++ b/test/e2e_node/node_problem_detector_linux.go
@@ -28,7 +28,6 @@ import (
 	"github.com/onsi/ginkgo"
 	"github.com/onsi/gomega"
 	v1 "k8s.io/api/core/v1"
-	rbacv1 "k8s.io/api/rbac/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/fields"
 	"k8s.io/apimachinery/pkg/labels"
@@ -77,9 +76,10 @@ var _ = framework.KubeDescribe("NodeProblemDetector [NodeFeature:NodeProblemDete
 			condition = v1.NodeConditionType("TestCondition")
 
 			// File paths used in the test.
-			logFile      = "/log/test.log"
-			configFile   = "/config/testconfig.json"
-			etcLocaltime = "/etc/localtime"
+			logFile        = "/log/test.log"
+			configFile     = "/config/testconfig.json"
+			kubeConfigFile = "/config/kubeconfig"
+			etcLocaltime   = "/etc/localtime"
 
 			// Volumes used in the test.
 			configVolume    = "config"
@@ -154,6 +154,28 @@ var _ = framework.KubeDescribe("NodeProblemDetector [NodeFeature:NodeProblemDete
 				]
 			}`
 
+			// This token is known to apiserver and its group is `system:masters`.
+			// See also the function `generateTokenFile` in `test/e2e_node/services/apiserver.go`.
+			kubeConfig := fmt.Sprintf(`
+apiVersion: v1
+kind: Config
+users:
+- name: node-problem-detector
+  user:
+    token: %s
+clusters:
+- cluster:
+    server: %s
+    insecure-skip-tls-verify: true
+  name: local
+contexts:
+- context:
+    cluster: local
+    user: node-problem-detector
+  name: local-context
+current-context: local-context
+`, framework.TestContext.BearerToken, framework.TestContext.Host)
+
 			ginkgo.By("Generate event list options")
 			selector := fields.Set{
 				"involvedObject.kind":      "Node",
@@ -166,35 +188,9 @@ var _ = framework.KubeDescribe("NodeProblemDetector [NodeFeature:NodeProblemDete
 			ginkgo.By("Create config map for the node problem detector")
 			_, err = c.CoreV1().ConfigMaps(ns).Create(context.TODO(), &v1.ConfigMap{
 				ObjectMeta: metav1.ObjectMeta{Name: configName},
-				Data:       map[string]string{path.Base(configFile): config},
-			}, metav1.CreateOptions{})
-			framework.ExpectNoError(err)
-
-			ginkgo.By("Create the service account for node problem detector")
-			_, err = f.ClientSet.CoreV1().ServiceAccounts(f.Namespace.Name).Create(context.TODO(), &v1.ServiceAccount{
-				ObjectMeta: metav1.ObjectMeta{
-					Name:      name,
-					Namespace: ns,
-				},
-			}, metav1.CreateOptions{})
-			framework.ExpectNoError(err)
-
-			ginkgo.By("Bind the cluster role system:node-problem-detector with the service account for node problem detector")
-			_, err = f.ClientSet.RbacV1().ClusterRoleBindings().Create(context.TODO(), &rbacv1.ClusterRoleBinding{
-				ObjectMeta: metav1.ObjectMeta{
-					Name: name,
-				},
-				RoleRef: rbacv1.RoleRef{
-					APIGroup: rbacv1.GroupName,
-					Kind:     "ClusterRole",
-					Name:     "system:node-problem-detector",
-				},
-				Subjects: []rbacv1.Subject{
-					{
-						Kind:      rbacv1.ServiceAccountKind,
-						Name:      name,
-						Namespace: ns,
-					},
+				Data: map[string]string{
+					path.Base(configFile):     config,
+					path.Base(kubeConfigFile): kubeConfig,
 				},
 			}, metav1.CreateOptions{})
 			framework.ExpectNoError(err)
@@ -264,7 +260,9 @@ var _ = framework.KubeDescribe("NodeProblemDetector [NodeFeature:NodeProblemDete
 							Args: []string{
 								"--logtostderr",
 								fmt.Sprintf("--system-log-monitors=%s", configFile),
-								fmt.Sprintf(" --apiserver-override=%s?inClusterConfig=true", framework.TestContext.Host),
+								// `ServiceAccount` admission controller is disabled in node e2e tests, so we could not use
+								// inClusterConfig here.
+								fmt.Sprintf("--apiserver-override=%s?inClusterConfig=false&auth=%s", framework.TestContext.Host, kubeConfigFile),
 							},
 							Env: []v1.EnvVar{
 								{
@@ -436,10 +434,6 @@ var _ = framework.KubeDescribe("NodeProblemDetector [NodeFeature:NodeProblemDete
 			gomega.Expect(e2epod.WaitForPodToDisappear(c, ns, name, labels.Everything(), pollInterval, pollTimeout)).To(gomega.Succeed())
 			ginkgo.By("Delete the config map")
 			c.CoreV1().ConfigMaps(ns).Delete(context.TODO(), configName, metav1.DeleteOptions{})
-			ginkgo.By("Delete the service account")
-			c.CoreV1().ServiceAccounts(ns).Delete(context.TODO(), name, metav1.DeleteOptions{})
-			ginkgo.By("Delete the clusterRoleBinding")
-			c.RbacV1().ClusterRoleBindings().Delete(context.TODO(), name, metav1.DeleteOptions{})
 			ginkgo.By("Clean up the events")
 			gomega.Expect(c.CoreV1().Events(eventNamespace).DeleteCollection(context.TODO(), *metav1.NewDeleteOptions(0), eventListOptions)).To(gomega.Succeed())
 			ginkgo.By("Clean up the node condition")

